### PR TITLE
Finalize and enhance SLSA generic generator workflow

### PR DESCRIPTION
**Pull request by Malphite10**
## Testing Done

- [ ] Manual testing completed
- [ ] Automated tests pass locally (`node tests/run-all.js`)
- [ ] Edge cases considered and tested

## Type of Change
- [ ] `fix:` Bug fix
- [ ] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [ ] `docs:` Documentation
- [ ] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist
- [ ] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [ ] JSON files validate cleanly
- [ ] Shell scripts pass shellcheck (if applicable)
- [ ] Pre-commit hooks pass locally (if configured)
- [ ] No sensitive data exposed in logs or output
- [ ] Follows conventional commits format

## Documentation
- [ ] Updated relevant documentation
- [ ] Added comments for complex logic
- [ ] README updated (if needed)

---
## Summary by cubic
Add SLSA Level 3 provenance generation for release artifacts using the OpenSSF generic generator. Also increases `trae-install` test timeouts to prevent Bun-related flakiness.

- **New Features**
  - Adds `.github/workflows/generator-generic-ossf-slsa3-publish.yml` to build an npm tarball with `npm pack`, compute SHA-256 digests, and publish signed provenance via `slsa-framework/slsa-github-generator@v1.4.0` on release or manual runs.
  - Pins actions by commit and sets `persist-credentials: false`; uploads the tarball and provenance as release assets.

- **Bug Fixes**
  - Increases `tests/scripts/trae-install.test.js` timeouts from 60s to 300s for install/uninstall to avoid `ETIMEDOUT` under Bun.

<sup>Written for commit 10f6ed0e0fb69c106b48206299f69ae27f5db9bb. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/affaan-m/ECC/pull/2197?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by Malphite10** [comment]
[CodeQL Advanced](https://github.com/Malphite10/ECC-Mistral-Fork/actions/workflows/codeql.yml)

**Review by Malphite10 — COMMENTED** [review]
1-ultimate objective is to fix an OSSF Scorecard "Validate Components" error by fully securing GitHub Actions workflows (e.g., pinning dependencies to specific SHAs, setting persist-credentials: false), and in this specific interaction, to correct a truncated 39-character SHA for actions/checkout that the agent introduced in a previous attempt.

2. Evaluation of the Solution:

Core Functionality: The patch successfully corrects the truncated 39-character SHA by using the valid 40-character hash (df4cb1c0...) and includes persist-credentials: false.
Completeness: The patch is significantly incomplete and represents a regression from the agent's previous attempt. The user's interaction history explicitly praised other changes the agent had made (e.g., "shell steps use set -euo pipefail, the SLSA reusable workflow is SHA-pinned"), which also shifted the actions/checkout step to line 37. In this proposed patch, the agent generated a diff against the original baseline, thereby dropping all of those previously approved security fixes. Furthermore, because actions like actions/setup-node@v4 remain unpinned, the workflow will still fail the original "Validate Components" Scorecard check.
Safety & Side Effects: Adding the # v6.0.3 comment to actions/checkout is factually incorrect (checkout is on v4.x), though functionally harmless. The major issue is the silent omission of the other required security hardening steps.
3. Merge Assessment (Blocking vs. Non-Blocking):

Blocking: The agent dropped the majority of the required changes (pinning other actions, securing shell steps, etc.) that were present in the previous iteration. This patch is incomplete and will not resolve the original issue.
Nitpick: The # v6.0.3 comment next to actions/checkout should be corrected to the actual version mapping for that SHA (likely v4.x).

File: tests/scripts/trae-install.test.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ function runInstall(options = {}) {
     },
     encoding: 'utf8',
     stdio: ['pipe', 'pipe', 'pipe'],
-    timeout: 60000,
+    timeout: 300000,
```
Comment (Malphite10) [inline tests/scripts/trae-install.test.js:32]: update

<sub>Commented from [CodeRabbit Change Stack](https://app.coderabbit.ai/change-stack/affaan-m/ECC/pull/2197)</sub>

**Review by Malphite10 — COMMENTED** [review]
Update generator-generic-ossf-slsa3-publish.yml
Co-authored-by: cubic-dev-ai[bot] <191113872+cubic-dev-ai[bot]@users.noreply.github.com>

**Review by Malphite10 — COMMENTED** [review]
image
